### PR TITLE
[TEST] deleteRecord test for serializer encapsulation side quest

### PR DESCRIPTION
**Pull request by Gaurav0**
For #6460

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -24,6 +24,7 @@
     "test:try-one": "yarn workspace ember-data test:try-one",
     "test:docs": "yarn workspace ember-data test:docs",
     "test:encapsulation": "lerna run --scope='*-encapsulation-test-app' test",
+    "test:encapsulation:serializer": "yarn workspace serializer-encapsulation-test-app test",
```
Comment (runspired) [inline package.json]: for local dev you can run `ember test` in the directory directly. For CI the `test:encapsulation` command already runs this. Let's remove this.

File: packages/-serializer-encapsulation-test-app/tests/integration/delete-record-test.js [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+    let _payloads = [
+      {
+        id: '1',
+        type: 'person',
+        attributes: {
+          firstName: 'John',
+          lastName: 'Smith',
+        },
+      },
+      '',
```
Comment (runspired) [inline packages/-serializer-encapsulation-test-app/tests/integration/delete-record-test.js]: if we leave a payload here for the `deleteRecord` response we do in fact call `normalizeResponse`. Let's add a test for that as well.

I added an issue for discussing the fact we don't always call `normalizeResponse` here: https://github.com/emberjs/data/issues/6541

I was not aware of this and it seems a bit bonkers.

Comment (Gaurav0) [reply packages/-serializer-encapsulation-test-app/tests/integration/delete-record-test.js]: I am confused about how it makes any sort of sense to have a response to a DELETE request and why we should do anything with it if there is. JSON-API I believe requires a 204 No Response.

File: package.json (hunk 1 above)
Comment (Gaurav0) [reply package.json]: done.

File: packages/-serializer-encapsulation-test-app/tests/integration/delete-record-test.js (hunk 2 above)
Comment (Gaurav0) [reply packages/-serializer-encapsulation-test-app/tests/integration/delete-record-test.js]: Added additional test.

File: packages/-serializer-encapsulation-test-app/tests/integration/delete-record-test.js [hunk 3]
```diff
@@ -0,0 +1,251 @@
...
+      type: this.constructor.modelName,
+      attributes: {
+        firstName,
+        lastName,
+      },
+    };
+  }
+}
+
+module('inteegration/delete-record - running deleteRecord with minimum serializer', function(hooks) {
```
Comment (runspired) [inline packages/-serializer-encapsulation-test-app/tests/integration/delete-record-test.js]:
```suggestion
module('integration/delete-record - running deleteRecord with minimum serializer', function(hooks) {
```

Comment (runspired) [reply packages/-serializer-encapsulation-test-app/tests/integration/delete-record-test.js]: I've fixed this spelling error in most the already merged tests, suspect you are copying from one we haven't fixed yet

**Pull request merged**